File: cmd/code/repeatedSubstringPattern.js

function repeatedSubstringPattern ( s )
{
    const doubleString = s + s;

    return doubleString.slice(1, doubleString.length - 1).includes(s)
}

console.log(repeatedSubstringPattern("abab"));
console.log(repeatedSubstringPattern("aba")); 
console.log(repeatedSubstringPattern("abcabcabcabc"));

File: cmd/code/repeatedSubstringPattern.go
package code

import (
	"strings"
)

func repeatedSubstringPattern(s string) bool {
	doubleString := s + s

	return  strings.Contains(doubleString[1:len(doubleString)-1],s)
}

// func main() {
// 	fmt.Println(repeatedSubstringPattern("abab"))       // true
// 	fmt.Println(repeatedSubstringPattern("aba"))        // false
// 	fmt.Println(repeatedSubstringPattern("abcabcabcabc")) // true
// }